code/modules/clothing/glasses/glasses.dm

```diff
@@ -51,6 +52,16 @@ BLIND     // can't see anything
 	normal_layer = !normal_layer
 	to_chat(usr, SPAN_NOTICE("\The [src] will now layer [normal_layer ? "under" : "over"] your hair."))
 
+/obj/item/clothing/glasses/protects_eyestab(var/stabbed = FALSE)
```

Comment:
I feel like this should be a (high) chance based on the weapon's force and not just an instant smash through.